.github/workflows/release.yml

```diff
@@ -0,0 +1,35 @@
+name: goreleaser
+
+on:
+  push:
+    # run only against stable semver tags on main
```

Comment:
Q: Should we do this strictly by the tag to support tagging releases on branches that aren't `main`?

Reply:
Did you have a specific use-case in mind where we'd cut a release off a branch other than `main`? If not, I'd stick with main.

Reply:
Mostly just thinking that "releases can only be created from commits on `main`" is an artificial constraint that may be overly restrictive.

Reply:
I think a strict, consistent release process is a good constraint. We can always revisit it later.

Reply:
Switched to triggering on `release` instead of `push` so that there's no ambiguity.